File: NativeAPI/Python/b2_download_file_by_id.py
import requests

api_url = ""  # Provided by b2_authorize_account
account_authorization_token = ""  # Provided by b2_authorize_account
file_id = ""  # The fileId of the file you want to download

response = requests.get(
    f"{api_url}/b2api/v4/b2_download_file_by_id",
    headers={"Authorization": account_authorization_token},
    params={"fileId": file_id}
)
content = response.text
print(content)

File: NativeAPI/Java/B2DownloadFileById.java
package com.backblaze.examples;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class B2DownloadFileById {
    public static void main(String[] args) {
        String apiUrl = ""; // Provided by b2_authorize_account
        String accountAuthorizationToken = ""; // Provided by b2_authorize_account
        String fileId = "";  // The file ID of the file you want to download

        try (HttpClient client = HttpClient.newHttpClient()) {
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create(apiUrl + "/b2api/v4/b2_download_file_by_id?fileId=" + fileId))
                    .header("Authorization", accountAuthorizationToken)
                    .GET()
                    .build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            String downloadedData = response.body();
            System.out.println(downloadedData);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
